### PR TITLE
Ben Ghalami - Potluck

### PR DESCRIPTION
**Pull request by bghalami**

### CODE REVIEW

**Comment by bghalami** [comment]
"frick" is my real product

File: lib/dish.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require "pry"
+
+class Dish
+  attr_reader :name, :category
+  def initialize(name, category)
+    @name = name
+    @category = category
```
Comment (s-espinosa) [inline lib/dish.rb]: 👍

File: lib/potluck.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    @dishes << dish
+  end
+
+  def get_all_from_category(category)
+    category_dishes = @dishes.map do |dish|
+      if dish.category == category
+        dish
+      end
+    end
+    category_dishes.compact
```
Comment (s-espinosa) [inline lib/potluck.rb]: 👍 Think this is a perfectly good straightforward approach.

Take a look at `select` as a potential alternative as well!

File: lib/potluck.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def menu
+    hash = Hash.new{0}
+    x = @dishes.map do |dish|
+      dish.category
+    end
+
+    x.map do |category|
```
Comment (s-espinosa) [inline lib/potluck.rb]: Why use map here? Not using the return.

File: lib/potluck.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      category_dishes = @dishes.map do |dish|
+        if dish.category == category
+          dish.name
+        end
+      end
+      category_dishes.compact
+      hash[category] = (category_dishes.compact).sort
+    end
+    hash
```
Comment (s-espinosa) [inline lib/potluck.rb]: Interesting! Like that you were able to power through this!

Assume that the main concern here is how to create keys and values. One thing to consider is iterating over the array of dishes, then doing a check to see if a key for that dish exists. If so, go ahead and add to it. If not, create the key value pair.

File: lib/potluck.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def ratio(category)
+    type = (get_all_from_category(category).length).to_f
+    total = (@dishes.length).to_f
+
+    (type / total) * 100
```
Comment (s-espinosa) [inline lib/potluck.rb]: 👍

File: test/potluck_test.rb [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    assert_equal [], potluck.dishes
+  end
+
+  def test_dishes_can_add
+    potluck = Potluck.new("7-13-18")
+    couscous_salad = Dish.new("Couscous Salad", :appetizer)
+    potluck.add_dish(couscous_salad)
+
+    refute potluck.dishes.empty?
```
Comment (s-espinosa) [inline test/potluck_test.rb]: This works, but can also actually make an assertion about the contents of the `dishes` array.

```
assert_equal [couscous_salad], potluck.dishes
```

File: test/potluck_test.rb [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+    potluck.add_dish(couscous_salad)
+    potluck.add_dish(roast_pork)
+    potluck.add_dish(cocktail_meatballs)
+    potluck.add_dish(summer_pizza)
+
+    expected = [couscous_salad,summer_pizza]
+    actual = potluck.get_all_from_category(:appetizer)
+
+    assert_equal expected, actual
```
Comment (s-espinosa) [inline test/potluck_test.rb]: 🎉

**Pull request closed without merging**